pipeline/extract.py

```diff
@@ -8,33 +9,62 @@
-    return datetime.datetime.fromtimestamp(time_in_ms/1000, tz=datetime.timezone.utc).minute
+    if not isinstance(time_in_ms, int):
+        raise TypeError("The input time_in_ms must be an integer.")
+
+    if time_in_ms < 0:
+        raise ValueError("The input time_in_ms cannot be negative.")
+
+    full_date = datetime.datetime.fromtimestamp(
+        time_in_ms / 1000, tz=datetime.timezone.utc)
+    return full_date.strftime("%H:%M")
 
 
 def get_all_earthquake_data(data_url: str) -> list[dict]:
     """Gets all earthquake data for the hour from USGS"""
-    response = requests.get(data_url, timeout=30)
-    data = response.json()
-    return data[FEATURES]
-
+    try:
+        response = requests.get(data_url, timeout=30)
+        response.raise_for_status()
+        data = response.json()
+        if FEATURES not in data:
+            raise KeyError(f"Expected key '{FEATURES}' not found in the response")
+        return data[FEATURES]
+    except requests.exceptions.Timeout as e:
+        print(f"Timeout occurred in get_all_earthquake_data: {e}")
+        return []
+    except requests.exceptions.RequestException as e:
+        print(f"RequestException occurred in get_all_earthquake_data: {e}")
+        return []
 
+      
 def get_current_earthquake_data(all_earthquake_data: list[dict]) -> list[dict]:
     """Gets all the most recent earthquakes from data"""
-    current_minute = datetime.datetime.now().minute
-    latest_earthquakes = []
-
-    for earthquake in all_earthquake_data:
-        print(get_minute_from_epoch_time(earthquake[PROPERTIES][TIME]))
-        if get_minute_from_epoch_time(earthquake[PROPERTIES][TIME]) == current_minute:
-            latest_earthquakes.append(earthquake)
+    try:
+        current_time = datetime.datetime.now(
+            tz=datetime.timezone.utc).strftime("%H:%M")
+        latest_earthquakes = []
 
-    return latest_earthquakes
+        for earthquake in all_earthquake_data:
+            if PROPERTIES in earthquake and TIME in earthquake[PROPERTIES]:
+                if get_time_from_epoch_time(earthquake[PROPERTIES][TIME]) == current_time:
+                    latest_earthquakes.append(earthquake)
+            else:
+                continue
+        return latest_earthquakes
+    except Exception as e:
+        print(f"Unexpected error occurred in get_current_earthquake_data: {e}")
+        return latest_earthquakes
 
 
 def extract_process() -> list[dict]:
-    """Runs the functions to extract all the latest earthquakes"""
-    all_data = get_all_earthquake_data(URL)
-    relevant_data = get_current_earthquake_data(all_data[FEATURES])
-    return relevant_data
+    """Runs the functions to extract all relevant data"""
+    try:
+        all_data = get_all_earthquake_data(URL)
+        relevant_data = get_current_earthquake_data(all_data)
+        return relevant_data
+    except Exception as e:
+        print("Error occurred in the extract process: " + e)
```

Comment:
```suggestion
        print("Error occurred in the extract process: " + str(e))
```
small note, you used f strings on line 57 but not here